src/main/dynamo/Table.ts

```diff
@@ -42,11 +44,56 @@ export class Table<PK extends string = string, SK extends string = string> {
     return new GSIBuilder(this, name)
   }
 
-  addToEncryptionBlacklist(field: string) {
-    this.encryptionBlacklist.add(field)
+  registerGSI(gsi: GSI<string, string, any>) {
+    this.gsis.push(gsi)
+    this.encryptionBlacklist.add(gsi.partitionKeyName)
+    this.encryptionBlacklist.add(gsi.sortKeyName)
   }
 
   getEncryptionBlacklist(): Set<string> {
     return this.encryptionBlacklist
   }
+
+  asCreateTableInput(
+    billingMode: DynamoDB.Types.BillingMode
+  ): DynamoDB.Types.CreateTableInput {
+    const attributeSet = new Set([
+      this.partitionKeyName,
+      this.sortKeyName,
+      "model",
+      ...this.gsis.flatMap((_) => [_.partitionKeyName, _.sortKeyName]),
+    ])
+
+    const attributeDefinitions: DynamoDB.Types.AttributeDefinitions = Array.from(
+      attributeSet
+    ).map((attr) => ({
+      AttributeName: attr,
+      AttributeType: "S",
```

Comment:
There's an existing assumption in Beyonce that all key/indexed attributes are strings. Would be nice to make it smarter there eventually.